src/util/union_find_replace.cpp

```diff
@@ -8,13 +8,13 @@ Author: Romain Brenguier, romain.brenguier@diffblue.com
 
 #include "union_find_replace.h"
 
-/// Keeps a map of symbols to expressions, such as none of the mapped values
-/// exist as a key
-/// \param a: an expression of type char array
-/// \param b: an expression to map it to, which should be either a symbol
-///             a string_exprt, an array_exprt, an array_of_exprt or an
-///             if_exprt with branches of the previous kind
-/// \return the new mapped value
+/// Merge the set containing `a` and the set containing `b`.
+/// It is guaranteed that if `b` never appears as the first argument of a
```

Comment:
Does this function provide that guarantee, or require it from the caller?

Reply:
I think that guarantee is too obscure/non-standard to provide.
To make anything the root, use re_root()

Reply:
I removed this part of the comment, because changing the implementation may break this.